HLTrigger/JetMET/plugins/HLTHFJetCleaner.cc

```diff
@@ -0,0 +1,269 @@
+// -*- C++ -*-
+//
+// Package:    HLTrigger/HLTHFJetCleaner
+// Class:      HLTHFJetCleaner
+//
+/**\class HLTHFJetCleaner HLTHFJetCleaner.cc HLTrigger/HLTHFJetCleaner/plugins/HLTHFJetCleaner.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Alp Akpinar
+//         Created:  Wed, 01 Dec 2021 15:47:19 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/METReco/interface/MET.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+//
+// class declaration
+//
+
+class HLTHFJetCleaner : public edm::global::EDProducer<> {
+public:
+  typedef std::vector<reco::Jet> JetCollection;
+  explicit HLTHFJetCleaner(const edm::ParameterSet&);
+  ~HLTHFJetCleaner() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+
+  template <typename T>
+  std::vector<T> extractValueMapParameters(edm::Handle<edm::ValueMap<T>> valueMapHandle, size_t numJets) const;
+
+  edm::EDGetTokenT<edm::View<reco::Jet>> jetToken_;
+  edm::EDGetTokenT<edm::View<reco::MET>> metToken_;
+
+  edm::EDGetTokenT<edm::ValueMap<float>> sigmaEtaEtaToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> sigmaPhiPhiToken_;
+  edm::EDGetTokenT<edm::ValueMap<int>> centralEtaStripSizeToken_;
+
+  // Minimum jet pt to consider while applying HF cuts
+  const float jetPtThresh_;
+  const float dphiJetMetThresh_;
+
+  // Cuts will be applied to jets only with jetEtaMin_ < |eta| < jetEtaMax_
+  const float jetEtaMin_;
+  const float jetEtaMax_;
+
+  const float sigmaEtaPhiDiffThresh_;
+  const float centralEtaStripSizeThresh_;
+  const float cornerCutSigmaEtaEta_;
+  const float cornerCutSigmaPhiPhi_;
+  const bool applyCornerCut_;
+  const bool applySigmaEtaPhiCut_;
+  const bool applyStripSizeCut_;
+
+  //virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  //virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+  //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
```

Comment:
Remove commented-out lines

Reply:
Implemented in [65bcbcc](https://github.com/cms-sw/cmssw/pull/37539/commits/65bcbccb4d1e7fe0a8b6b2ea2d8835d03d8b6e99)